xss_helpers.go

```diff
@@ -38,104 +33,104 @@ func isBlackAttr(s string) int {
 		return attributeTypeNone
 	}
 
+	upperS := strings.ToUpper(strings.ReplaceAll(s, "\x00", ""))
 	if length >= 5 {
 		// javascript on.*
 		if strings.ToUpper(s[:2]) == "ON" {
 			// got javascript on- attribute name
 			return attributeTypeBlack
 		}
 
-		if strings.ToUpper(strings.ReplaceAll(s, "\x00", "")) == "XMLNS" ||
-			strings.ToUpper(strings.ReplaceAll(s, "\x00", "")) == "XLINK" {
+		if upperS == "XMLNS" || upperS == "XLINK" {
 			// got xmlns or xlink tags
 			return attributeTypeBlack
 		}
 	}
 
 	for _, black := range blacks {
-		if strings.ToUpper(strings.ReplaceAll(s, "\x00", "")) == black.name {
+		if upperS == black.name {
 			// got banner attribute name
 			return black.attributeType
 		}
 	}
 	return attributeTypeNone
 }
 
-func htmlDecodeByteAt(s string, consumed *int) int {
+func htmlDecodeByteAt(s string, consumed int) (int, int) {
 	length := len(s)
 	val := 0
 
 	if length == 0 {
-		*consumed = 0
-		return byteEOF
+		consumed = 0
```

Comment:
`consumed` is initialized to either `0` or `1` no matter what, so it seems like it previously was literally an output variable only. I guess we can remove the input parameter

Reply:
Also if I'm not mistaken, we could remove the `consumed` local variable too to make the logic easier to follow